zarr/core.py

```diff
@@ -423,18 +423,31 @@ def __array__(self, *args):
             a = a.astype(args[0])
         return a
 
-    def __iter__(self):
+    def islice(self, start=None, end=None):
```

Comment:
There should probably be some range checking on ``start`` and ``end`` here. I guess we should follow the interface of ``itertools.islice`` as closely as possible and return the same errors they do on different out-of-range conditions.

Reply:
I played around a little with `islice`. Writing this down here so i don't forget:

- Setting a bound exceeding the array bound silently returns the array without warning or error.
- Bounds must be non negative integers (or None), otherwise a ValueError is raised.
- end > start returns [], reverse iteration is not supported at all.

I think that should cover it, or am I forgetting something?